superset/models/core.py

```diff
@@ -706,15 +703,35 @@ def needs_conversion(df_series):
                 return True
             return False
 
-        for k, v in df.dtypes.items():
-            if v.type == numpy.object_ and needs_conversion(df[k]):
-                df[k] = df[k].apply(utils.json_dumps_w_dates)
-        return df
+        with closing(engine.raw_connection()) as conn:
+            with closing(conn.cursor()) as cursor:
+                for sql in sqls:
+                    self.db_engine_spec.execute(cursor, sql)
+                df = pd.DataFrame.from_records(
+                    data=list(cursor.fetchall()),
```

Comment:
This LGTM. There's a bit of complexity/differences still left around the way explore vs sqllab get their data. I think the main difference is around Hive, the `async` param and whether `handle_cursor` is called or not (called only in SQL Lab).

This PR is a step in the right direction. Maybe eventually we want to make bring more into db_engine_spec, and have a `get_records` that receives `update_process=False` that internally would deal with `async`.

Reply:
@mistercrunch do you want me to makes these changes here, or should these issues be addressed in a future PR?